src/docfx/build/page/BuildPage.cs

```diff
@@ -377,5 +382,10 @@ private static string CreateContent(Context context, Document file, JObject page
             ValidateBookmarks(context, file, htmlDom);
             return CreateHtmlContent(context, htmlDom);
         }
+
+        private static bool IsCustomized404Page(Document file)
+        {
+            return Path.GetFileName(file.FilePath.Path).Equals("404.md", PathUtility.PathComparison);
```

Comment:
Keep the original behavior, cause a user could write a 404.yml and published to 404

Reply:
For v2, only the `404.md` is treated as a customized 404 page. do we want to keep the behavior?

Reply:
That was a bug (hole) in v2 🤣 ,

Reply:
Got you, the `404.md` and `404.yml` will both be consumed as 404 file in DHS, I will do the change